### PR TITLE
[FLINK-14962] fix findbugs WMI_WRONG_MAP_ITERATOR  warning

### PR DESCRIPTION
**Pull request by caidezhi**
## What is the purpose of the change

This PR fixes WMI_WRONG_MAP_ITERATOR warnings reported by FindBugs.

## Brief change log
use an iterator on the entrySet of the map, to avoid the Map.get(key) lookup.
related class :
org.apache.flink.formats.parquet.ParquetMapInputFormat
org.apache.flink.graph.library.CommunityDetection
org.apache.flink.streaming.connectors.fs.bucketing.BucketingSink (already marked as "Deprecated")

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? ( no)

### CODE REVIEW

**Comment by caidezhi** [comment]
@flinkbot run travis

**Comment by caidezhi** [comment]
@flinkbot run travis

File: flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java [hunk 1]
```diff
@@ -893,11 +892,10 @@ private void handlePendingFilesForPreviousCheckpoints(Map<Long, List<String>> pe
 
 		LOG.debug("Moving pending files to final location on restore.");
 
-		Set<Long> pastCheckpointIds = pendingFilesPerCheckpoint.keySet();
-		for (Long pastCheckpointId : pastCheckpointIds) {
-			// All the pending files are buckets that have been completed but are waiting to be renamed
-			// to their final name
-			for (String filename : pendingFilesPerCheckpoint.get(pastCheckpointId)) {
+		for (Map.Entry<Long, List<String>> entry : pendingFilesPerCheckpoint.entrySet()) {
+			Long pastCheckpointId = entry.getKey();
```
Comment (KurtYoung) [inline flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java]: Please keep the comment.

Comment (caidezhi) [reply flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java]: Kurt, thanks for reviewing. I add back the original comments as you suggested.

**Pull request merged**